src/views/virtualmachines/details/tabs/details/components/grid/rightGrid/VirtualMachineDetailsRightGridLayout.tsx

```diff
@@ -52,6 +58,25 @@ const VirtualMachineDetailsRightGridLayout: React.FC<VirtualMachineDetailsRightG
       }),
     [],
   );
+
+  const vmWorkload = getWorkload(vm);
+
+  const updateWorkload = async (newWorkload: WORKLOADS) => {
```

Comment:
you need the async function? async will always return Promise, you can simply return k8sPatch